src/main/java/com/ibm/g11n/pipeline/iam/TokenLifeCylceManager.java

```diff
@@ -0,0 +1,248 @@
+/*
+ * Copyright IBM Corp. 2019
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.ibm.g11n.pipeline.iam;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.io.UnsupportedEncodingException;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.TimeUnit;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+
+/**
+ * Manages IAM token for the given IAM API key. By default a stored token will
+ * be considered expired when 85% of its valid time has passed. To change this
+ * default value, property IAM_TOKEN_EXPIRY_THRESHOLD can be set between
+ * 0.1(excluding) and 1(excluding). For example, for token to be considered
+ * expired when 75% of its valid time has passed, set it to 0.75.
+ * 
+ * @author Siddharth Jain
+ *
+ */
+public class TokenLifeCylceManager implements TokenManager {
```

Comment:
typo in class name? Should it be `TokenLifeCycleManager`?

Reply:
yes, corrected.